### PR TITLE
[script.transmission] 1.0.2

### PR DESCRIPTION
**Pull request by basilgello**
### Description

Version 1.0.2

* Fix torrent progress bar in main screen (thanks @AnonTester)
* Fix error with missing ETA on stalled torrents (thanks @AnonTester)
* Fix KeyError on torrent removal
* Add localization for new strings

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

### CODE REVIEW

File: script.transmission/resources/lib/gui.py [hunk 1]
```diff
@@ -168,10 +198,13 @@ def onClick(self, controlID):
         if (controlID == 112):
             # Remove selected torrent
             item = list.getSelectedItem()
-            if item and xbmcgui.Dialog().yesno(common.get_localized_string(32000),
-                                               'Remove \'%s\'?' % self.torrents[int(item.getProperty('TorrentID'))].name):
-                remove_data = xbmcgui.Dialog().yesno(common.get_localized_string(32000), 'Remove data as well?')
-                self.transmission.remove(int(item.getProperty('TorrentID')), remove_data)
+            if item and xbmcgui.Dialog().yesno(
+                    common.get_localized_string(32000), 'Remove \'%s\'?' %
```
Comment (pkscout) [inline script.transmission/resources/lib/gui.py:202]: This string needs to be localized as well.

File: script.transmission/resources/lib/gui.py [hunk 2]
```diff
@@ -168,10 +198,13 @@ def onClick(self, controlID):
...
+                    self.torrents[int(item.getProperty('TorrentID'))].name):
+                remove_data = xbmcgui.Dialog().yesno(
+                    common.get_localized_string(32000), 'Remove data as well?')
```
Comment (pkscout) [inline script.transmission/resources/lib/gui.py:205]: This string needs to be localized as well.

File: script.transmission/resources/lib/gui.py [hunk 3]
```diff
@@ -232,32 +274,47 @@ def close(self):
...
+
+        # Catch KeyError which indicates the torrent has been removed by user
+        try:
+            torrent = self.transmission.info()[self.torrent_id]
+        except KeyError:
+            return
+
         files = self.transmission.get_files(self.torrent_id)[self.torrent_id]
 
         statusline = "[%(status)s] %(down)s down, %(up)s up (Ratio: %(ratio).2f)" % \
```
Comment (pkscout) [inline script.transmission/resources/lib/gui.py:308]: If statusline is used displayed to the user in the GUI, it should also be localized.

File: script.transmission/resources/lib/transmissionrpc/constants.py [hunk 4]
```diff
@@ -1,295 +1,572 @@
...
+# set +- <argument1> - [<type>, <added version>, <removed version>, <previous argument name>, <next argument name>, <description>]
+#  |  +- <argument2> - [<type>, <added version>, <removed version>, <previous argument name>, <next argument name>, <description>]
+#  |
+# get +- <argument1> - [<type>, <added version>, <removed version>, <previous argument name>, <next argument name>, <description>]
+#     +- <argument2> - [<type>, <added version>, <removed version>, <previous argument name>, <next argument name>, <description>]
+
+# Arguments for torrent methods
+TORRENT_ARGS = {
+    'get': {
+        'activityDate': ('number', 1, None, None, None,
```
Comment (pkscout) [inline script.transmission/resources/lib/transmissionrpc/constants.py:66]: I'm assuming the strings here are only used internally, but if they are displayed to the end user via the GUI, they need to be localized.

**Review by pkscout — CHANGES_REQUESTED** [review]

**Comment by enen92** [comment]
Ping @basilgello

**Comment by basilgello** [comment]
I see, gonna fix it after Kodi trunk builds here

File: script.transmission/resources/lib/transmissionrpc/constants.py (hunk 4 above)
Comment (basilgello) [reply script.transmission/resources/lib/transmissionrpc/constants.py:66]: These are JSON-RPC method arguments :)

**Review by enen92 — APPROVED** [review]

**Pull request merged**